cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc

```diff
@@ -150,9 +152,18 @@ bool SubstraitToVeloxPlanValidator::validateScalarFunction(
   if (name == "extract") {
     return validateExtractExpr(params);
   }
-  if (name == "regexp_extract_all" && !scalarFunction.arguments()[1].value().has_literal()) {
-    logValidateMsg("native validation failed due to: pattern is not literal for regex_extract_all.");
-    return false;
+  if (name == "regexp_extract_all" || name == "regexp_extract" || name == "regexp_replace" || name == "rlike") {
```

Comment:
nit: There are 10 spark build-in functions related to regex, it is better to use `list.contains` to define whether to check pattern.

Reply:
Could you provide those function names or hint me where to find them? I only found these 4 functions.

Reply:
We are making split to support regex, so split also needs to be added here later.

Reply:
> Here is the list for functions that starts with regexp. https://spark.apache.org/docs/latest/api/sql/index.html#regexp And, `rlike` and `split` use regex too.

The url points to the latest spark function list. Many functions are not supported in spark 3.3.2 https://spark.apache.org/docs/3.3.2/api/sql/index.html#regexp_extract

Currently there should still be the 4 functions. `regex`, `regex_like` and `rlike` are the same and all aliased to `rlike.
`split` is in blacklist. I suppose there's no missing functions.

Reply:
Okey, we can change it after `split` support regex with re2.